src/hotspot/os/windows/os_windows.cpp

```diff
@@ -3241,18 +3242,37 @@ static size_t large_page_init_decide_size() {
     return 0;
   }
 
-#if defined(IA32) || defined(AMD64)
-  if (size > 4*M || LargePageSizeInBytes > 4*M) {
-    WARN("JVM cannot use large pages bigger than 4mb.");
-    return 0;
-  }
+  // Check if OS version is 11 or greater
+  if (win32::is_windows_11_or_greater() || win32::is_windows_server_2022_or_greater()) {
+      // OS-specific logic for versions 11 or greater
+      // You can implement specific checks or logic here for newer OS versions
+}
+  else {
+      // Existing logic for OS versions less than 11
+#if defined(IA32)
+      if (size > 4 * M || LargePageSizeInBytes > 4 * M) {
+          WARN("JVM cannot use large pages bigger than 4mb.");
+          return 0;
+      }
+#elif defined(AMD64)
+      if (!EnableAllLargePageSizes) {
```

Comment:
Do we expect that a user would ever want to specify EnableAllLargePageSizes to be true on an older Windows system? What do we intend to be the behavior in that case?

Reply:
In that instance we should issue a warning stating the level of support and that the flag is being ignored

Reply:
added